### PR TITLE
Add Fantom specific modification to be used with Sonic

### PR DESCRIPTION
**Pull request by thaarok**
This includes changes necessary for using in Sonic.

### CODE REVIEW

File: p2p/discover/discfilter/discfilter.go
Comment (HerbertJordan) [inline p2p/discover/discfilter/discfilter.go:1]: Should the operations in this file be thread safe?

File: p2p/enr/enr.go [hunk 1]
```diff
@@ -144,6 +144,19 @@ func (r *Record) Load(e Entry) error {
 	return &KeyError{Key: e.ENRKey(), Err: errNotFound}
 }
 
+func (r *Record) Has(key string) bool {
```
Comment (HerbertJordan) [inline p2p/enr/enr.go:147]: nit: `Has` => `HasKey` or `ContainsKey`?

File: p2p/server.go [hunk 2]
```diff
@@ -830,6 +835,10 @@ func (srv *Server) postHandshakeChecks(peers map[enode.ID]*Peer, inboundCount in
 	case c.node.ID() == srv.localnode.ID():
 		return DiscSelf
 	default:
+		if !c.is(trustedConn) && discfilter.Banned(c.node.ID(), c.node.Record()) && rand.Intn(5) != 0 {
+			// rarely accept useless peers
+			return DiscTooManyPeers
```
Comment (HerbertJordan) [inline p2p/server.go:840]: Should we have a dedicated error for "useless" peers?

File: params/version.go [hunk 3]
```diff
@@ -20,32 +20,32 @@ import (
...
-var VersionWithMeta = func() string {
-	v := Version
+func VersionWithMeta() string {
+	v := Version()
 	if VersionMeta != "" {
 		v += "-" + VersionMeta
 	}
 	return v
-}()
+}
```
Comment (HerbertJordan) [inline params/version.go:42]: Are these changes needed?

File: p2p/server.go (hunk 2 above)
Comment (thaarok) [reply p2p/server.go:840]: Maybe... but we may want to hide this reason from the peer

File: params/version.go (hunk 3 above)
Comment (thaarok) [reply params/version.go]: Reworked to not need version mutability in ethereum at all...

File: p2p/discover/discfilter/discfilter.go
Comment (thaarok) [reply p2p/discover/discfilter/discfilter.go:1]: "dynamic" LRU is thread safe, "enabled" boolean is written only during the app initialization, so at least for the current implementation it does not need to be thread-safe

**Review by HerbertJordan — APPROVED** [review]
Thanks for spending extra time in minimising those changes.

We will have to rebase this every now and then in the future. Making the changes as small as possible will help us with this.
